benches/benches/transaction_throughput.rs

```diff
@@ -0,0 +1,286 @@
+    },
+    fuel_vm::checked_transaction::{
+        CheckPredicateParams,
+        EstimatePredicates,
+    },
+};
+use futures::{
+    stream::FuturesUnordered,
+    FutureExt,
+    StreamExt,
+};
+use rand::{
+    rngs::StdRng,
+    SeedableRng,
+};
+use std::{
+    sync::Arc,
+    time::Duration,
+};
+use test_helpers::builder::{
+    TestContext,
+    TestSetupBuilder,
+};
+
+// Use Jemalloc during benchmarks
+#[global_allocator]
+static GLOBAL: tikv_jemallocator::Jemalloc = tikv_jemallocator::Jemalloc;
+
+fn bench_txs<F>(group_id: &str, c: &mut Criterion, f: F)
+where
+    F: Fn(&mut StdRng) -> Script,
+{
+    let inner_bench = |c: &mut BenchmarkGroup<WallTime>, n: u64| {
+        let id = format!("{}", n);
+        c.bench_function(id.as_str(), |b| {
+            let rt = tokio::runtime::Builder::new_current_thread()
+                .enable_all()
+                .build()
+                .unwrap();
+            let _drop = rt.enter();
+
+            let mut rng = rand::rngs::StdRng::seed_from_u64(2322u64);
+
+            let mut transactions = vec![];
+            for _ in 0..n {
+                transactions.push(f(&mut rng));
+            }
+
+            let mut test_builder = TestSetupBuilder::new(2322);
+            // setup genesis block with coins that transactions can spend
+            test_builder.config_coin_inputs_from_transactions(
+                &transactions.iter().collect::<Vec<_>>(),
+            );
+            // disable automated block production
+            test_builder.trigger = Trigger::Never;
+            test_builder.utxo_validation = true;
+
+            // spin up node
+            let transactions: Vec<Transaction> =
+                transactions.into_iter().map(|tx| tx.into()).collect();
+            let transactions = Arc::new(transactions);
+
+            b.to_async(&rt).iter_custom(|iters| {
+                let mut elapsed_time = Duration::default();
+                let test_builder = test_builder.clone();
+                let transactions = transactions.clone();
+
+                async move {
+                    for _ in 0..iters {
+                        let mut test_builder = test_builder.clone();
+                        let transactions = transactions.clone();
+                        // start the node
+                        let TestContext {
+                            srv: _dont_drop,
+                            client,
+                            ..
+                        } = test_builder.finalize().await;
+
+                        // insert all transactions
+                        let submits = FuturesUnordered::new();
+                        for tx in transactions.iter() {
+                            submits.push(client.submit(&tx).boxed());
+                        }
+                        let _: Vec<_> = submits.collect().await;
+                        let start = std::time::Instant::now();
+                        let _ = client.produce_blocks(1, None).await;
```

Comment:
> Interestingly the throughput difference between noop predicates and eck1 verification is fairly minimal, indicating the bottlenecks are in other areas besides tx validity checking (i.e. db performance). However the results show that a fuel-core node can easily process roughly 20k tps at full throttle without any p2p or block timing restrictions.

It is because you try to bench it without validation of transaction inputs. After the https://github.com/FuelLabs/fuel-core/pull/1249, we've started to reuse `Checked<Tx>` from TxPool. Because in the benchmark, you don't measure the insertion into the TxPool. You only measure the execution of the empty transaction.

I would say that it is better to bench the block validation because optimized block production still doesn't show the capability of the whole network.